pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -479,7 +478,15 @@ pub(crate) async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
                     Ok(Ok(file_size)) => {
                         usage_assumed.add_available_bytes(file_size);
                     }
-                    Ok(Err((file_size, EvictionError::NotFound | EvictionError::Downloaded))) => {
+                    Ok(Err((
+                        file_size,
+                        Some(EvictionError::NotFound | EvictionError::Downloaded),
+                    ))) => {
+                        evictions_failed.file_sizes += file_size;
+                        evictions_failed.count += 1;
+                    }
+                    Ok(Err((file_size, None))) => {
+                        // count timeouted as failed evictions even if they might complete later
```

Comment:
I would strongly prefer an enum that spells out the three different Ok() cases.

Just look at the type of `next`, way too much nesting IMO.

* `Ok(Ok(_))` => `EvictLayerResult::Succeeded`
* `Ok(Err(_, Some(_)))` => `EvictLayerResult::Failed`
* `Ok(Err(_, None))` => `EvictLayerResult::EvictCallTimedOut`

Reply:
Also, the treatment of `Ok(Err(_, Some()))` and `Ok(Err(_, None))` seems to be the same?

Reply:
> Also, the treatment of Ok(Err(_, Some())) and Ok(Err(_, None)) seems to be the same?

Per description yes that is the case. I think I wanted to duplicate them, but perhaps it does no good.

Reply:
> I would strongly prefer an enum that spells out the three different Ok() cases.

Aah yes, join error wrapping. Ok, I see what you mean.

Reply:
I'll probably push the timeout to `Layer::evict_and_wait` and so I'll have it for both this and eviction_task.

Reply:
Pushed, and added a new grandiose "smoke_test" which "tests" this, though quite leniently.